File: py/IEC61970/Base/Protection/CurrentRelay.py
# Converted by an OPENAI API call using model: gpt-3.5-turbo-1106
from CIM_STD_PYTHON.TC57CIM.IEC61970.Base.Domain.CurrentFlow import CurrentFlow
from CIM_STD_PYTHON.TC57CIM.IEC61970.Base.Domain.Seconds import Seconds
from CIM_STD_PYTHON.TC57CIM.IEC61970.Base.Protection.ProtectionEquipment import ProtectionEquipment


class CurrentRelay(ProtectionEquipment):
    def __init__(self):
        super().__init__()
        self.current_limit1 = CurrentFlow()  # * Current limit number one 1 for inverse time pickup.
        self.current_limit2 = CurrentFlow()  # * Current limit number 2 for inverse time pickup.
        self.current_limit3 = CurrentFlow()  # * Current limit number 3 for inverse time pickup.
        self.inverse_time_flag = False  # * Set true if the current relay has inverse time characteristic.
        self.time_delay1 = Seconds()  # * Inverse time delay number 1 for current limit number 1.
        self.time_delay2 = Seconds()  # * Inverse time delay number 2 for current limit number 2.
        self.time_delay3 = Seconds()  # * Inverse time delay number 3 for current limit number 3.

File: java/IEC61970/Base/Protection/CurrentRelay.java
package IEC61970.Base.Protection;

import IEC61970.Base.Domain.CurrentFlow;
import IEC61970.Base.Domain.Boolean;
import IEC61970.Base.Domain.Seconds;

/**
 * A device that checks current flow values in any direction or designated
 * direction.
 * @created 25-Dec-2023 8:31:55 PM
 */
public class CurrentRelay extends ProtectionEquipment {

	/**
	 * Current limit number one 1 for inverse time pickup.
	 */
	public CurrentFlow currentLimit1;
	/**
	 * Current limit number 2 for inverse time pickup.
	 */
	public CurrentFlow currentLimit2;
	/**
	 * Current limit number 3 for inverse time pickup.
	 */
	public CurrentFlow currentLimit3;
	/**
	 * Set true if the current relay has inverse time characteristic.
	 */
	public Boolean inverseTimeFlag;
	/**
	 * Inverse time delay number 1 for current limit number 1.
	 */
	public Seconds timeDelay1;
	/**
	 * Inverse time delay number 2 for current limit number 2.
	 */
	public Seconds timeDelay2;
	/**
	 * Inverse time delay number 3 for current limit number 3.
	 */
	public Seconds timeDelay3;

	public CurrentRelay(){

	}
}//end CurrentRelay